include/pybind11/pybind11.h

```diff
@@ -132,8 +132,8 @@ class cpp_function : public function {
                                         ? &rec->data : rec->data[0]);
 
             /* Override policy for rvalues -- always move */
-            constexpr auto is_rvalue = !std::is_pointer<Return>::value
-                                       && !std::is_lvalue_reference<Return>::value;
+            constexpr auto is_rvalue = detail::none_of<std::is_pointer<Return>,
+                                                       std::is_lvalue_reference<Return>>::value;
```

Comment:
Though I wonder if it wouldn't be better replaced with: `constexpr auto is_rvalue = !detail::move_never<Return>::value;`